### ISSUE TITLE
cv2.imshow "The function is not implemented. Rebuild the library" 

### ISSUE THREAD

**Issue by Crescent-Saturn** [skvark/opencv-python#18]
OpenCV Error: Unspecified error (The function is not implemented. Rebuild the library with Windows, GTK+ 2.x or Carbon support. If you are on Ubuntu or Debian, install libgtk2.0-dev and pkg-config, then re-run cmake or configure script) in cvShowImage, file /io/opencv/modules/highgui/src/window.cpp, line 545

please, OpenCV version :3.1.0
Python version: 2.7.12
OS: Elementary OS Loki.

Thanks!

**Comment by skvark** [comment]
Duplicate of https://github.com/skvark/opencv-python/issues/17

**Comment by hphari1998** [comment]
getting same error
please help..
in-advance thank you

**Comment by hg2051** [comment]
It's because of ''opencv-python-headless'. Uninstall it!

**Comment by Mayur8991** [comment]
ya  it is because of pip install OpenCV- python- headless just uninstall it and your error will be sorted

**Comment by dwaine18** [comment]
uninstalled opencv-python-headless and still the same error. I am using Ubuntu.

**Comment by nofreewill42** [comment]
1. `sudo apt install libgtk2.0-dev pkg-config` as the prompt says for Ubuntu users ; but same error
2. `pip uninstall opencv-python-headless` ; other error
3. `pip uninstall opencv-python; pip install opencv-python` ; [Solved]

**Comment by Yuval46412** [comment]
I have the same problem too. I did pip uninstall opencv-python-headless and it still does not work.

**Comment by Kitty1122** [comment]
Tried pip uninstall opencv-python-headless - didn't work
Also, pip uninstall opencv-python; pip install open-python - didn't work

**Comment by TracyLoza** [comment]
> I have the same problem too. I did pip uninstall opencv-python-headless and it still does not work.

Same to me, but I do this:
`pip uninstall opencv-python`
Then:
` pip install opencv-python`

**Comment by bkaan99** [comment]
Solve: Unistall opencv-python-headless

**Comment by LikhithChakravarthi** [comment]
I have tried everything, unistalling opencv-python, contrib modules and reinstalling them and Uninstalling headless module, but not working. Can ayone help me?

**Comment by bkaan99** [comment]
@LikhithChakravarthi  Hi Addinational Solution :
Use **Pycharm**,
Install **opencv-python** and **cv2** from the module installation location of the interpreter you will use from the **interpreter options in Pycharm**.

**Comment by obensustam2** [comment]
Please use Pycharm, it is way easier to manage the packages

**Comment by venkat1230** [comment]
Hay 
please try to update the -->  { python -m pip install opencv-python --upgrade } <-- in windows
it worked for me.

**Comment by Sellouk** [comment]
The error you're encountering indicates that the version of OpenCV you're using was built without GUI support for window management. To resolve this issue and display the camera feed in a window, you can either rebuild OpenCV with GUI support or use an alternative library to display the camera feed. Here are two possible solutions:

Solution 1: Rebuild OpenCV with GUI Support (Recommended)

Ensure you have CMake installed on your system.

Rebuild OpenCV with GUI support using CMake. You can do this by following these steps:

Download the OpenCV source code from the official OpenCV website: https://opencv.org/releases/

Extract the downloaded source code archive to a directory.

Open a terminal or command prompt and navigate to the directory where you extracted the OpenCV source code.

Create a build directory within the OpenCV source directory and navigate to it:
Configure the build using CMake with GUI support enabled:
using : 
cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local -D BUILD_opencv_python2=OFF -D BUILD_opencv_python3=ON -D WITH_TBB=ON -D WITH_V4L=ON -D WITH_QT=ON -D WITH_OPENGL=ON ..

After successful configuration, build and install OpenCV:
make
sudo make install



Once the installation is complete, you should have OpenCV with GUI support.

Update your Python environment to use the newly built OpenCV library. You may need to uninstall the previous OpenCV package and install the new one.

**Comment by Lelouch2503** [comment]
> Hay please try to update the --> { python -m pip install opencv-python --upgrade } <-- in windows it worked for me.

Thanks, it worked for me too=)))

**Comment by vivekmorariya** [comment]
if you are using spyder then ...
uninstall the open cv which you have installed  by command prompt or anaconda prompt 
then install the opencv library using the terminal available in the spyder itself 
then try again to run the same code/program using spyder it will run.

**Comment by VanshRana12** [comment]
> The error you're encountering indicates that the version of OpenCV you're using was built without GUI support for window management. To resolve this issue and display the camera feed in a window, you can either rebuild OpenCV with GUI support or use an alternative library to display the camera feed. Here are two possible solutions:
> 
> Solution 1: Rebuild OpenCV with GUI Support (Recommended)
> 
> Ensure you have CMake installed on your system.
> 
> Rebuild OpenCV with GUI support using CMake. You can do this by following these steps:
> 
> Download the OpenCV source code from the official OpenCV website: https://opencv.org/releases/
> 
> Extract the downloaded source code archive to a directory.
> 
> Open a terminal or command prompt and navigate to the directory where you extracted the OpenCV source code.
> 
> Create a build directory within the OpenCV source directory and navigate to it: Configure the build using CMake with GUI support enabled: using : cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local -D BUILD_opencv_python2=OFF -D BUILD_opencv_python3=ON -D WITH_TBB=ON -D WITH_V4L=ON -D WITH_QT=ON -D WITH_OPENGL=ON ..
> 
> After successful configuration, build and install OpenCV: make sudo make install
> 
> Once the installation is complete, you should have OpenCV with GUI support.
> 
> Update your Python environment to use the newly built OpenCV library. You may need to uninstall the previous OpenCV package and install the new one.

To do this but for windows:
After running this command successfully:
"cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local -D BUILD_opencv_python2=OFF -D BUILD_opencv_python3=ON -D WITH_TBB=ON -D WITH_V4L=ON -D WITH_QT=ON -D WITH_OPENGL=ON .."

Instead of executing "make" and "sudo make install" as these are bash commands,
In powershell you have to execute these commands in the build directory:
**To Build:** cmake --build . --config Release
**To Install:** cmake --build . --target INSTALL

Now uninstall and then reinstall "opencv-python" using pip.

Done!

**Comment by tegarpf90** [comment]
pip uninstall opencv-python; pip install opencv-python  Solved by pycharm

**Comment by hzx829** [comment]
On Ubuntu, this work for me:
```bash
sudo apt-get install python3-opencv
```

**Comment by oussamaJMAA** [comment]
> Hay please try to update the --> { python -m pip install opencv-python --upgrade } <-- in windows it worked for me.

ugprading to a newer version solved the issue for me ( opencv 4.9.0)

**Comment by DAVID-Hown** [comment]
@skvark 
@hzx829 
@nofreewill42 
@hzx829 
```
# apt install libgtk2.0-dev pkg-config
Reading package lists... Done
Building dependency tree       
Reading state information... Done
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 libgl1-mesa-glx : Depends: libgl1 but it is not going to be installed
                   Depends: libglx-mesa0 but it is not going to be installed
 libgtk2.0-dev : Depends: libgtk2.0-0 (= 2.24.32-4ubuntu4) but it is not going to be installed
                 Depends: gir1.2-gtk-2.0 (= 2.24.32-4ubuntu4) but it is not going to be installed
                 Depends: libgtk2.0-common but it is not going to be installed
                 Depends: libglib2.0-dev (>= 2.27.3) but it is not going to be installed
                 Depends: libgdk-pixbuf2.0-dev (>= 2.21.0) but it is not going to be installed
                 Depends: libpango1.0-dev (>= 1.20) but it is not going to be installed
                 Depends: libatk1.0-dev (>= 1.29.2) but it is not going to be installed
                 Depends: libcairo2-dev (>= 1.6.4-6.1) but it is not going to be installed
                 Depends: libx11-dev (>= 2:1.0.0-6) but it is not going to be installed
                 Depends: libxext-dev (>= 1:1.0.1-2) but it is not going to be installed
                 Depends: libxinerama-dev (>= 1:1.0.1-4.1) but it is not going to be installed
                 Depends: libxi-dev (>= 1:1.0.1-4) but it is not going to be installed
                 Depends: libxrandr-dev (>= 2:1.2.99) but it is not going to be installed
                 Depends: libxcursor-dev but it is not going to be installed
                 Depends: libxfixes-dev (>= 1:3.0.0-3) but it is not going to be installed
                 Depends: libxcomposite-dev (>= 1:0.2.0-3) but it is not going to be installed
                 Depends: libxdamage-dev (>= 1:1.0.1-3) but it is not going to be installed
                 Depends: libxml2-utils but it is not going to be installed
                 Recommends: debhelper but it is not going to be installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
```
```
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "/usr/local/lib/python3.8/dist-packages/cv2/qt/plugins" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb.
```
```
import cv2
image = cv2.imread("1.jpg")
cv2.imshow("image", image)
cv2.waitKey(0)
cv2.destroyAllWindows()
```
When I connect to the server remotely using pycharm, I want a local visualization
I have encountered this problem, I would like to ask you how to solve it？

https://stackoverflow.com/questions/64784315/qt-qpa-xcb-could-not-connect-to-displayerror-while-using-opencv-on-ec2-ubuntu-s： In my past experience, pycharm can do local visualization, but sometimes it will go wrong, is this related to the pycharm version or is it related to the remote server

**Comment by jmhasan1** [comment]
> It's because of ''opencv-python-headless'. Uninstall it!

Hie , I have uninstall opencv-headless but upon uninstalling that I am unable to import opencv 
ModuleNotFoundError: No module named 'cv2'
(upon uninstalling opencv-headless)
And if I reinstall opencv (with opencv-headless) , it is imported but stuck in same error

**Comment by c2727c** [comment]
Did uninstall then re-install opencv-python and it worked for me.

before:
   - opencv-python==4.8.0.74
   - opencv-python-headless==4.9.0.80
after:
    - opencv-python==4.9.0.80
    - opencv-python-headless==4.9.0.80
 
So I guess this error can be caused by mismatched versions  of opencv-python and opencv-python-headless.

**Comment by dineth88** [comment]
If you are using Pycharm, just simply change the interpreter and try to run your application again. It solves my problem relating with this error.

**Comment by haileyesus34** [comment]
Uninstalling the opencv-python and then reinstalling it worked for me

**Comment by anu-rrn1** [comment]
> Did uninstall then re-install opencv-python and it worked for me.
> 
> before:
> 
> * opencv-python==4.8.0.74
> * opencv-python-headless==4.9.0.80
>   after:
>   - opencv-python==4.9.0.80
>   - opencv-python-headless==4.9.0.80
> 
> So I guess this error can be caused by mismatched versions of opencv-python and opencv-python-headless.

YES,
STEP 1: pip uninstall opencv-python opencv-contrib-python opencv-python-headless -y

(Answered by Chatgpt, when asked the use of this)
This command removes all installed versions of OpenCV (opencv-python, opencv-contrib-python, and opencv-python-headless) from your system.
It's often done to clean up the environment before installing a specific version or resolving compatibility issues.

STEP 2: pip install opencv-python==4.9.0.80

**WORKED FOR ME**
THANK YOU

**Comment by rachmatuty** [comment]
you just need to reinstall the regular version of opencv not the headless version that is already installed in the folder. you must enter the envi folder that has been activated first. then uninstall the opencv that has been installed in the folder with the code: (py -m pip uninstall opencv-python-headless). then enter. after that type y and enter again. then install the regular open cv with the code: (py -m pip install opencv-python).

**Comment by zgjja** [comment]
For anyone who uses the docker images come from NVIDIA NGC container, you should uninstall headless-opencv, reinstall opencv-python, besides you need to remove the following folder manually:
```bash
rm -rf /usr/local/lib/python3.12/dist-packages/cv2
```

**Note:** change the python version above to the one in your container
